Sources/Interaction/Style/InteractorStyleUnicam/index.js

```diff
@@ -0,0 +1,497 @@
+
+    const delta = {
+      x: normalizedPosition.x - normalizedPreviousPosition.x,
+      y: normalizedPosition.y - normalizedPreviousPosition.y,
+    };
+
+    const camera = renderer.getActiveCamera();
+    const cameraPosition = camera.getPosition();
+
+    const movec = [];
+    for (let index = 0; index < 3; index++) {
+      movec[index] = model.downPoint[index] - cameraPosition[index];
+    }
+
+    const offset1 = [];
+    for (let index = 0; index < 3; index++) {
+      offset1[index] = movec[index] * delta.y * -4;
+    }
+
+    model.translateCamera(camera, offset1);
+
+    const { rightV } = model.getRightVAndUpV(model.downPoint, camera);
+
+    const offset2 = [];
+    for (let index = 0; index < 3; index++) {
+      offset2[index] = delta.x * rightV[index];
+    }
+
+    model.translateCamera(camera, offset2);
+
+    renderer.resetCameraClippingRange();
+    publicAPI.modified();
+  };
+
+  //----------------------------------------------------------------------------
+  model.normalize = (position) => {
+    const size = publicAPI.getInteractor().getView().getSize();
+
+    const [width, height] = size;
+
+    const nx = -1.0 + (2.0 * position.x) / width;
+    const ny = -1.0 + (2.0 * position.y) / height;
+
+    return { x: nx, y: ny };
+  };
+
+  model.getRightVAndUpV = (downPoint, camera) => {
+    const cameraPosition = camera.getPosition();
+    const cameraToPointVec = [0, 0, 0];
+
+    for (let index = 0; index < 3; index++) {
+      cameraToPointVec[index] = downPoint[index] - cameraPosition[index];
+    }
+
+    const atV = camera.getViewPlaneNormal();
+    vtkMath.normalize(atV);
+    const l = vtkMath.dot(cameraToPointVec, atV);
+    const viewAngle = (camera.getViewAngle() * vtkMath.Pi()) / 180.0;
+    const [width, height] = publicAPI.getInteractor().getView().getSize();
+
+    const scaleX = (width / height) * ((2 * l * Math.tan(viewAngle / 2)) / 2);
+    const scaleY = (2 * l * Math.tan(viewAngle / 2)) / 2;
+
+    const upV = camera.getViewUp();
+    const rightV = [];
+    vtkMath.cross(upV, atV, rightV);
+    vtkMath.cross(atV, rightV, upV);
+    vtkMath.normalize(rightV);
+    vtkMath.normalize(upV);
+
+    for (let index = 0; index < 3; index++) {
+      rightV[index] *= scaleX;
+      upV[index] *= scaleY;
+    }
+
+    return { rightV, upV };
+  };
+
+  //----------------------------------------------------------------------------
+  model.translateCamera = (camera, offset) => {
+    const cameraPosition = camera.getPosition();
+    const cameraFocalPoint = camera.getFocalPoint();
+    const newCameraPosition = [];
+    const newCameraFocalPoint = [];
+
+    for (let index = 0; index < 3; index++) {
+      newCameraPosition[index] = cameraPosition[index] + offset[index];
+      newCameraFocalPoint[index] = cameraFocalPoint[index] + offset[index];
+    }
+
+    camera.setPosition(...newCameraPosition);
+    camera.setFocalPoint(...newCameraFocalPoint);
```

Comment:
optimize that by creating a new publicAPI.translate(x, y, z) method in vtkCamera